dio/test/request_integration_test.dart

```diff
@@ -0,0 +1,336 @@
+          response.data['headers']['Content-Type'],
+          Headers.jsonContentType,
+        );
+      });
+
+      test('DELETE', () async {
+        final response = await dio.deleteUri(
+          Uri.parse('/delete'),
+        );
+        expect(response.statusCode, 200);
+        expect(response.isRedirect, isFalse);
+        expect(response.data['method'], 'DELETE');
+      });
+
+      test('DELETE with content', () async {
+        final response = await dio.deleteUri(
+          Uri.parse('/delete'),
+          data: data,
+        );
+        expect(response.statusCode, 200);
+        expect(response.isRedirect, isFalse);
+        expect(response.data['method'], 'DELETE');
+        expect(response.data['json'], data);
+        expect(
+          response.data['headers']['Content-Type'],
+          Headers.jsonContentType,
+        );
+      });
+    });
+
+    group('redirects', () {
+      test('single', () async {
+        final response = await dio.get(
+          '/redirect',
+          queryParameters: {'url': 'https://httpbun.com/get'},
+          onReceiveProgress: (received, total) {
+            // ignore progress
+          },
+        );
+        expect(response.isRedirect, isTrue);
+
+        if (!isWeb) {
+          // Redirects are not supported in web.
+          // Rhe browser will follow the redirects automatically.
+          expect(response.redirects.length, 1);
+          final ri = response.redirects.first;
+          expect(ri.statusCode, 302);
+          expect(ri.location.path, '/get');
+          expect(ri.method, 'GET');
+        }
+      });
+
+      test('multiple', () async {
+        final response = await dio.get(
+          '/redirect/3',
+        );
+        expect(response.isRedirect, isTrue);
+
+        if (!isWeb) {
+          // Redirects are not supported in web.
+          // Rhe browser will follow the redirects automatically.
+          expect(response.redirects.length, 3);
+          final ri = response.redirects.first;
+          expect(ri.statusCode, 302);
+          expect(ri.method, 'GET');
+        }
+      });
+    });
+
+    group('status codes', () {
+      for (final code in [400, 401, 404, 500, 503]) {
+        test('$code', () async {
+          expect(
+            dio.get('/status/$code').catchError(
+                (e) => throw (e as DioException).response!.statusCode!),
+            throwsA(equals(code)),
+          );
+        });
+      }
+    });
+
+    test('multi value headers', () async {
+      final Response response = await dio.get(
+        '/get',
+        options: Options(
+          headers: {
+            'x-multi-value-request-header': ['value1', 'value2'],
+          },
+        ),
+      );
+      expect(response.statusCode, 200);
+      expect(response.isRedirect, isFalse);
+      expect(
+        response.data['headers']['X-Multi-Value-Request-Header'],
+        // TODO we have a diff here between browser and non-browser
+        equals('value1, value2'),
```

Comment:
Here we have different results on io & web, need to check what is going on.

Reply:
I adjusted the browser behaviour here, we were actually sending `List.toString()` which resulted in `[foo, bar]` formatting. Running the web request tests now also against `httpbun` detected this.